### DISCUSSION TITLE
Enum for scope

### DISCUSSION THREAD

**Discussion by EricGaim** [FortnoxAB/csharp-api-sdk#211] [Q&A] [answered]
Hello!

I have been working with this sdk to try helping a colleague have to do a little less menial work but I am still a novice so I'm having an issue. I was able to get everything working with static token but having to use enums for scope is giving me a little bit of an issue. I can't seem to find a way to get any good answers on the right way to do it. May I have some help on this please?

Kind regards,
Eric Wingård

**Comment by richardrandak** [comment, marked as answer]
Hello Eric. What exactly is the problem? 
The scopes are used only for the new auth workflow. The static token workflow does not use them.

Can you please describe your situation with a little more details?

**Reply by EricGaim** [reply]
Hello Richard!

The issue is when I'm trying to use the BuildAuthUri method I want to use 3 parameters. The client Id which is simply a string with my client id, the 2 scopes that I am planning on using and state which I use the GenerateState method to generate. I plan on leaving the uri parameter blank as from my understanding of reading the source code that is a possibility. My issue comes from using the scopes parameter. How should I write my code for that input as I can't find a way that does not give me an error. I'm assuming it's something simple I'm missing.

**Reply by richardrandak** [reply]
What error are you getting? Which scopes are you using? Are you getting an error when building the uri or when using it for getting the token? All the scopes you are requesting should be also enabled for the app in the developer's portal.

Well, here is a simple code which I assume you're doing. Is that right?

```
        var clientId = "myclientid";
        var scopes = new Scope[]
        {
            Scope.Customer,
            Scope.CostCenter,
            Scope.CompanyInformation
        };

        var authClient = new FortnoxAuthClient();
        var authWorkflow = authClient.StandardAuthWorkflow;

        var uri = authWorkflow.BuildAuthUri(clientId, scopes, authWorkflow.GenerateState());
```

**Reply by EricGaim** [reply]
I was having issue how to enter the scope parameter but that code example just solved my issue. I was having trouble with how to format the scope list the correct way.

Thanks!

**Reply by richardrandak** [reply]
You're welcome! Let me know in case of other issues.

The function takes IEnumerable<Scope>, so you can use whichever collection you prefer, the array was just an example

```
// var scopes = new Scope[]
var scopes = new List<Scope>()
```

**Reply by EricGaim** [reply]
Yeah that makes sense now. I thought the input had to be an enum. While I have you here do I understand the other discussions correctly with that if I have a Oauth1 solution before they discontinue it I can still use that after december 9th?

**Reply by richardrandak** [reply]
By Oauth1 you mean the old workflow with non-expirable token right?

1) The deadline was moved to January 9th - https://developer.fortnox.se/blog/delay-with-oauth2/
2) All tokens should still continue to work, but you won't be able to activate a new one. So, basically, you won't be able to get any new customers for that solution.

I can not guarantee you anything anymore, but this is what I heard from Fortnox support and how I understand it :)

**Reply by EricGaim** [reply]
Yes exacly, static token. Since this solution is only for us to automate our own process and are not looking to sell this to any customers it should be fine with static token, right?

**Reply by richardrandak** [reply]
Yes, it should be fine! You already have a static token right?

One thing to note here is that even though the deadline is not here yet, in my experience, new apps created after September have the old workflow disabled already, more specifically, they do not appear in the integrations list so the API key (auth code) can't be generated. It was not documented anywhere, but I had this problem.

**Reply by EricGaim** [reply]
I believe my app was created in oktober or november originally but I have a static token for it so I'm assuming it will be fine and won't suddenly stop working in January right? The problem right now is that the app is only being used in our test environment so I might have to try and get a static token working straight away on the production version.

**Reply by richardrandak** [reply]
Yes, please try it. In my situation, I could find the app by clientid for several minutes, but then it disappeared from the list, so later I could not reproduce it. So maybe you managed to get the token in that weird "window", which seems like a bug :D  Let me know how is it in your situation.
